sdks/wasm/Mono.WebAssembly.DebuggerProxy/MonoProxy.cs

```diff
@@ -432,6 +451,70 @@ async Task GetDetails(MessageId msg_id, MonoCommands cmd, CancellationToken toke
 			}
 
 		}
+		
+		
+		public async Task<JObject> GetVariableValue (MessageId msg_id, int scope_id, string expression, CancellationToken token)
+		{
+			JObject thisValue = null;
+			var context = GetContext (msg_id);
+			if (context.CallStack == null)
+				return null;
+			var scope = context.CallStack.FirstOrDefault (s => s.Id == scope_id);
+			var vars = scope.Method.GetLiveVarsAt (scope.Location.CliLocation.Offset);
+			//get_this
+			int i = 0;
+			int [] var_ids = { };
+			var res = await SendMonoCommand (msg_id, MonoCommands.GetScopeVariables (scope.Id, var_ids), token);
+			var values = res.Value? ["result"]? ["value"]?.Values<JObject> ().ToArray ();
+			for (; i < values.Length; i++) {
+				if (values [i] ["name"].Value<string> () == "this") {
+					thisValue = values [i];
+				}
+			}
+
+			if (thisValue != null && expression == "this") {
+				return thisValue;
+			}
+			//search in locals
+			var_ids = vars.Where (v => v.Name == expression).Select (v => v.Index).ToArray ();
+			if (var_ids.Length > 0) {
+				res = await SendMonoCommand (msg_id, MonoCommands.GetScopeVariables (scope.Id, var_ids), token);
+				values = res.Value? ["result"]? ["value"]?.Values<JObject> ().ToArray ();
+				return values [0];
+			}
+			//search in scope
+			if (thisValue != null) {
+				var objectId = thisValue ["value"] ["objectId"].Value<string> ();
+				var parts = objectId.Split (new char [] { ':' });
+				res = await SendMonoCommand (msg_id, MonoCommands.GetObjectProperties (int.Parse (parts [2]), expandValueTypes: false), token);
+				values = res.Value? ["result"]? ["value"]?.Values<JObject> ().ToArray ();
+				for (i = 0; i < values.Length; i++) {
+					if (values [i] ["name"].Value<string> () == expression) {
+						return values [i];
+					}
+				}
+			}
+			return null;
+		}
+
+		async Task GetEvaluateOnCallFrame (MessageId msg_id, int scope_id, string expression, CancellationToken token)
+		{
+			JObject thisValue = null;
+			var context = GetContext (msg_id);
+			if (context.CallStack == null)
+				return;
+
+			var varValue = await GetVariableValue (msg_id, scope_id, expression, token);
```

Comment:
Maybe `TryGetVariableValue`? It would make it more obvious that this might not be able to find the value.